lib/std/os/linux/tls.zig

```diff
@@ -7,56 +18,63 @@ const native_arch = @import("builtin").cpu.arch;
-// In this case we allocate additional space for our control structure that's
-// placed _before_ the DTV pointer together with the DTV.
-//
-// NOTE: Some systems such as power64 or mips use this variant with a twist: the
-// alignment is not present and the tp and DTV addresses are offset by a
-// constant.
-//
-// On the other hand the variant II has the following layout in memory:
-// ---------------------------------------
-// |  TLS  | TCB |     Zig     |   DTV   |
-// | block |     | thread data | storage |
-// --------^------------------------------
-//         `-- The thread pointer register points here
-//
-// The structure of the TCB is not defined by the ABI so we reserve enough space
-// for a single pointer as some architectures such as x86 and x86_64 need a
-// pointer to the TCB block itself at the address pointed by the tp.
-//
-// In this case the control structure and DTV are placed one after another right
-// after the TLS block data.
-//
-// At the moment the DTV is very simple since we only support static TLS, all we
-// need is a two word vector to hold the number of entries (1) and the address
-// of the first TLS block.
-//
-// [1] https://www.akkadia.org/drepper/tls.pdf
-
-const TLSVariant = enum {
-    VariantI,
-    VariantII,
+/// Represents an ELF TLS variant.
+///
+/// In all variants, the TP and the TLS blocks must be aligned to the `p_align` value in the
+/// `PT_TLS` ELF program header. Everything else has natural alignment.
+///
+/// The location of the DTV does not actually matter. For simplicity, we put it in the TLS area, but
+/// there is no actual ABI requirement that it reside there.
+const Variant = enum {
+    /// The original Variant I:
+    ///
+    /// ----------------------------------------
+    /// | DTV | Zig TCB | ABI TCB | TLS Blocks |
+    /// ----------------^-----------------------
+    ///                 `-- The TP register points here.
+    ///
+    /// The layout in this variant necessitates separate alignment of both the TP and the TLS
+    /// blocks.
+    ///
+    /// The first word in the ABI TCB points to the DTV. For some architectures, there may be a
+    /// second word with an unspecified meaning.
+    I_original,
+    /// The modified Variant I:
+    ///
+    /// ---------------------------------------------------
+    /// | DTV | Zig TCB | ABI TCB | [Offset] | TLS Blocks |
+    /// -------------------------------------^-------------
+    ///                                      `-- The TP register points here.
+    ///
+    /// The offset (which can be zero) is applied to the TP only; there is never physical gap
+    /// between the ABI TCB and the TLS blocks. This implies that we only need to align the TP.
+    ///
+    /// The first (and only) word in the ABI TCB points to the DTV.
+    I_modified,
+    /// Variant II:
+    ///
+    /// ----------------------------------------
+    /// | TLS Blocks | ABI TCB | Zig TCB | DTV |
+    /// -------------^---------^----------------
```

Comment:
Is the second `^` here a copy paste error? The TP register only point to one place, right?

Reply:
Not sure how exactly that ended up there, but yes, there should just be the one.